extensions/shared/external-media/sources/google-photos/index.js

```diff
@@ -0,0 +1,295 @@
+/**
+ * WordPress dependencies
+ */
+import apiFetch from '@wordpress/api-fetch';
+import { __ } from '@wordpress/i18n';
+import { Component, Fragment, __experimentalCreateInterpolateElement } from '@wordpress/element';
+import { SelectControl, Button, Path, SVG } from '@wordpress/components';
+
+/**
+ * Internal dependencies
+ */
+import { SOURCE_GOOGLE_PHOTOS, PATH_RECENT, PATH_ROOT } from '../../constants';
+import withMedia from '../with-media';
+import MediaBrowser from '../../media-browser';
+import requestExternalAccess from '../request-external-access';
+import { getApiUrl } from '../api';
+import GoogleFilterOption from './filter-option';
+import GoogleFilterView from './filter-view';
+import getFilterRequest from './filter-request';
+
+const isImageOnly = allowed => allowed && allowed.length === 1 && allowed[ 0 ] === 'image';
+
+class GooglePhotosMedia extends Component {
```

Comment:
As we move towards more consistency in the Jetpack repo (see #15694), I think we'll need to move away from classes in new code that's added to Jetpack.

Reply:
I have rewritten most of them now, the most complex ones are still left and will be done.